cli/src/cli.rs

```diff
@@ -1237,6 +1242,12 @@ fn do_process_deploy(
                 balance,
             ));
             balance_needed = balance;
+        } else if account.lamports > minimum_balance && !force_use_program_address {
```

Comment:
Let's check for `system_program::check_id(&account.owner)` here too.  If it's not a system account then there's nothing the user can do about it.